src/js/angular/guides/tour-lib-services/shepherd.service.js

```diff
@@ -0,0 +1,543 @@
+     *           <li>right</li>
+     *           <li>right-start</li>
+     *           <li>right-end</li>
+     *           <li>left</li>
+     *           <li>left-start</li>
+     *           <li>left-end</li>
+     *         </ul>
+     *     </li>
+     *     <li><b>type</b> - the type of step. Possible values:
+     *       <ul>
+     *           <li>
+     *               <b>clickable-element</b> - the element is clickable like link, button... Client of workbench can click on this element.
+     *           </li>
+     *           <li>
+     *               <b>read-only-element</b> - the element is readable. Client of workbench can only read this component.
+     *           </li>
+     *           <li>
+     *               <b>input-element</b> - the element is some kind ot input like text, text area ... . Client of workbench can enter data in the element.
+     *           </li>
+     *        </ul>
+     *     </li>
+     *     <li>
+     *         <b>maxWaitTime</b> - some element need more time to be present on the page. This is the max waiting time element to be visible
+     *         on the page. Default value is 3 second.
+     *     </li>
+     *     <li>
+     *         <b>canBePaused</b> - some element need user interaction with some other element on the page before appear. Step which uses such
+     *         element can be persist when guide is paused. For example if step is "Click on sub menu "Repositories" of menu "Setup""
+     *     </li>
+     *     <li>
+     *         <b>onNextClick</b> - function which will be executed when nex button is clicked.
+     *     </li>
+     *     <li>
+     *         <b>onPreviousClick</b> - function which will be executed when previous button is clicked.
+     *     </li>
+     *
+     * </ul>
+     * @param startStepId - start step id.
+     */
+    this.startGuide = (guideFileName, stepsDescriptions, startStepId) => {
+        if (!stepsDescriptions) {
+            return;
+        }
+        const guide = this._createGuide(guideFileName)
+        this._addGuideSteps(guide, stepsDescriptions);
+        this._startGuide(guide, startStepId);
+    }
+
+    /**
+     * Starts guide from the step where it was paused.
+     */
+    this.resumeGuide = (guideFileName, stepsDescriptions, startStepId) => {
+        LocalStorageAdapter.set(GUIDE_PAUSE, false);
+        this.startGuide(guideFileName, stepsDescriptions, startStepId);
+    }
+
+    /**
+     * Cancels currently started guide if any.
+     */
+    this.cancel = () => {
+        const activeGuide = Shepherd.activeTour;
+        if (activeGuide) {
+            activeGuide.cancel();
+        }
+        this._clearLocaleStorage();
+        if (!!this.onCancel) {
+            this.onCancel();
+        }
+    }
+
+    this.getGuideFileName = () => {
+        return LocalStorageAdapter.get(GUIDE_NAME);
+    }
+
+    this.getCurrentStepId = () => {
+        return LocalStorageAdapter.get(GUIDE_CURRENT_STEP_ID);
```

Comment:
I'm thinking something here. If I run a guide and pause it. Then someone else sits on the computer and logs in with another user, he/she will see an unfinished guide and can resume/terminate it. Right?

Reply:
Yes, you are right. Yes, you are right. This behaviour is same for everything: chosen repository, saved query in a tab.

Reply:
I know, but this doesn't look like a good UX to me. @desislava-hristova-ontotext @avataar Do we plan something about separating the data stored in the browser by the WB per user?